tests/explicit-nulls/neg-patmat/match-pat.scala

```diff
@@ -1,3 +1,5 @@
+// scalac: -Xfatal-warnings
```

Comment:
Not a zealot (much), but `-Werror` is more sleek and modern.
```
Possible warning options include:
         -Wconf  Configure compiler warnings.
        -Werror  Fail the compilation if there are any warnings.
```
My missed opportunity today on a scala 2 PR, for which I am overcompensating:

![image](https://github.com/lampepfl/dotty/assets/369425/eaf98de4-9362-4bcf-a38e-05982226753a)

Reply:
Given the size of this PR I prefer to keep all flags identical. We can refactor the flags after this PR.